### PR TITLE
[4.x]Upload and Update

### PR DESCRIPTION
**Pull request by brianteeman**
Change the link with long text that is apparently confusing people to shorter text that is styled like a button

Pull Request for Issue #37512 .

## After

image

### CODE REVIEW

**Comment by laoneo** [comment]
Is there a reason this button should not be inline and the same size with the other two ones?

**Comment by brianteeman** [comment]
yes absolutely. its not a preferred option hence it is smaller and below. just as the link text was before

File: administrator/components/com_joomlaupdate/tmpl/joomlaupdate/noupdate.php [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 if (Factory::getApplication()->getIdentity()->authorise('core.admin', 'com_joomlaupdate'))
 {
-	$displayData['formAppend'] = '<div class="text-center text-decoration-underline">' . HTMLHelper::_('link', $uploadLink, Text::_($displayData['textPrefix'] . '_EMPTYSTATE_APPEND')) . '</div>';
+	$displayData['formAppend'] = '<div class="text-center"><a href="' . $uploadLink . '" class="btn btn-sm btn-outline-secondary mx-auto">' . Text::_('COM_JOOMLAUPDATE_UPLOAD_AND_UPDATE') . '</a></div>';
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/tmpl/joomlaupdate/noupdate.php]:
```suggestion
	$displayData['formAppend'] = '<div class="text-center"><a href="' . $uploadLink . '" class="btn btn-sm btn-outline-secondary">' . Text::_('COM_JOOMLAUPDATE_UPLOAD_AND_UPDATE') . '</a></div>';
```

Comment (brianteeman) [reply administrator/components/com_joomlaupdate/tmpl/joomlaupdate/noupdate.php]: why?

Comment (Quy) [reply administrator/components/com_joomlaupdate/tmpl/joomlaupdate/noupdate.php]: `mx-auto` has no effect on inline element. Thus the use of `text-center`.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 2718226864379d06c02cc2132ed2b5ae22f1af0a<br /><br />Tested successfully in Joomla 4.2.3-dev of 8 April in Wampserver 3.2.7 using PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37519">issues.joomla.org/tracker/joomla-cms/37519</a>.</sub>

**Comment by Webdongle** [comment]
I have tested this item :white_check_mark: successfully on 2718226864379d06c02cc2132ed2b5ae22f1af0a<br /><br />Success<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37519">issues.joomla.org/tracker/joomla-cms/37519</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37519">issues.joomla.org/tracker/joomla-cms/37519</a>.</sub>

**Comment by Kostelano** [comment]
I am not a designer, I look at the result as a normal user. It looks terrible.
The string option was better.

**Comment by laoneo** [comment]
Adding here the RLDQ label.

**Comment by toivo** [comment]
ReleasLeadDecisionQueue because of one opinion to the contrary?

The word 'msnually' does not belong to the string. It causes confusion and it needs to go.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37519">issues.joomla.org/tracker/joomla-cms/37519</a>.</sub>

**Comment by Kostelano** [comment]
Just for thought. Why not put "Upload and Install" and "Reinstall" buttons in the toolbar? How do I get into the menu for reinstalling kernel files now? I can not.

Screenshot_1

**Comment by Webdongle** [comment]
> Just for thought. Why not put "Upload and Install" and "Reinstall" buttons in the toolbar? How do I get into the menu for reinstalling kernel files now? I can not.

That is a different issue.
btw
reinstall 01

Although tabs like in J3 would imho be better.  But the powers that be decided on this design.  Anyway it is the wording itself this is dealing with.  If you want to change the layout start another tracker and create a patch for that.

**Comment by simbus82** [comment]
> I am not a designer, I look at the result as a normal user. It looks terrible. The string option was better.

We are de facto in a "admin management app", for a good UX actions need to be identified by buttons.
A linked text is wrong in this type of UI, this isn't a webpage. (for example, it would make more sense to refer to an external page of the guide as linked text and not as a button).

**Comment by Kostelano** [comment]
@simbus82 
It doesn't matter to me - a button or a line - both can be a good solution, BUT. I'm talking about the fact that THIS visual solution loses to the original version. Perhaps if the button could be implemented in the same size/style, and also placed in a better place (it's hard for me to think of a better one, except for, for example, a toolbar), this would be a better solution.

In any case, the decision is up to the maintainers.

**Comment by Webdongle** [comment]
@Kostelano 
The position is in the same place as the incorrect text was.  The size of the button balances the page.  But it is not about the position it is about the fact that the wording needs to be changed to 'Upload and Update'.  Please stop derailing this thread by making off topic comments.  If you want the position changed please start another separate tracker.

**Comment by bembelimen** [comment]
I'm ok with the button but I have some doubts about the text. Probably it's a thing as I'm not a native speaker, but I'm missing the "what should I upload" information.

"Update by uploading the update package"

Something like that (although I'm still not 100% happy)?

**Comment by brianteeman** [comment]
It's the text that's causing the problems

**Comment by brianteeman** [comment]
All that information is on the next page

image

**Comment by Webdongle** [comment]
@bembelimen 
'Upload and Update' is the wording that has been used for at least 7 years.  Why wait until now to object to it's use?

'Upload and Update' is also used in the helpscreen linked to from the Component
https://help.joomla.org/proxy?keyref=Help41:Joomla_Update#Upload_and_Update
"**You can use this feature to update Joomla if your server is behind a firewall or otherwise unable to contact the update servers. First download the Joomla Upgrade Package in ZIP format from the official Joomla download page.**"

**Comment by bembelimen** [comment]
> All that information is on the next page
> 
> image

Fair enough, just we shouldn't remove existing language strings then I'm ok.

**Comment by brianteeman** [comment]
> Fair enough, just we shouldn't remove existing language strings then I'm ok.

I really disagree with keeping unused strings for core components. It makes no logical seense to me at all so to move this forward I have changed this pr to keep the strings with the new values

**Comment by bembelimen** [comment]
Not many rules, it's in the B/C promise... Thx

**Comment by brianteeman** [comment]
> Not many rules, it's in the B/C promise... Thx

Doesn't mean it makes sense ;)

**Comment by bembelimen** [comment]
Thx

**Comment by brianteeman** [comment]
thanks - got there in the end

**Comment by toivo** [comment]
Thank you everyone!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37519">issues.joomla.org/tracker/joomla-cms/37519</a>.</sub>

**Comment by Webdongle** [comment]
Thank you especially @brianteeman

**Pull request merged**